pkg/status/statuscollector.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+)
+
+func (c *Controller) syncStatusCollector(ctx context.Context, key string) error {
+
+	statusCollector, err := c.statusCollectorLister.Get(key)
+	if err != nil {
+		// The resource no longer exist, which means it has been deleted.
+		if errors.IsNotFound(err) {
+			utilruntime.HandleError(fmt.Errorf("object %#v in work queue no longer exists", key))
```

Comment:
This is not an error condition, it is something that can happen even when every user and bit of code is operating correctly. The code should not handle this as an error.

Reply:
Fixed

Reply:
Indeed.